File: android/app/src/main/java/com/kurt/bgmate/domain/repository/RecommendRepository.kt
package com.kurt.bgmate.domain.repository

import com.kurt.bgmate.domain.model.BoardGame
import com.kurt.bgmate.domain.model.RecommendCondition
import com.kurt.bgmate.domain.model.RecommendResult

interface RecommendRepository {
    suspend fun recommend(condition: RecommendCondition, includeNew: Boolean, ownedGames: List<BoardGame>): List<RecommendResult>
}

File: flutter/lib/domain/repository/recommend_repository.dart
import 'package:bgmate_flutter/domain/model/board_game.dart';
import 'package:bgmate_flutter/domain/model/recommend_condition.dart';
import 'package:bgmate_flutter/domain/model/recommend_result.dart';

abstract interface class RecommendRepository {
  Future<List<RecommendResult>> recommend({
    required RecommendCondition condition,
    required bool includeNew,
    required List<BoardGame> ownedGames,
  });
}
